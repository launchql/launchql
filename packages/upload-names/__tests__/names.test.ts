// @ts-nocheck
import { sync as glob } from 'glob';
import { basename } from 'path';

import getName from '../src';

jest.setTimeout(3000000);
const files = []
  .concat(glob(__dirname + '/../../../__fixtures__/kitchen-sink/**'))
  .concat(glob(__dirname + '/../../../__fixtures__/kitchen-sink/**/.*'))
  .filter((file) => {
    const key = file.split('kitchen-sink')[1];
    return key != '';
  })
  .concat([
    'niño.yo',
    'beça',
    'Selam Dünya.jpg',
    'Салом Ҷаҳон.gif',
    'Përshendetje Botë',
    'PËRSHENDETJE BOTË',
    'PËRSH- - - ENDETJE  BOTË'
  ]);

const allNonEnglish = [].concat([
  'どうもありがとうミスターロボット.txt',
  'Лев.р',
  'Дмитрий',
  '全国温泉ガイド.jp',
  '你好，世界',
  '你好，世界',
  '你好，世界.世界'
]);

describe('uploads', () => {
  it('english', async () => {
    const res = {};
    const use = files;
    for (let i = 0; i < use.length; i++) {
      const file = use[i];
      const results = getName(file);
      res[basename(file)] = results;
    }
    expect(res).toMatchSnapshot();
  });
  it('upper', async () => {
    const res = {};
    const use = files;
    for (let i = 0; i < use.length; i++) {
      const file = use[i];
      const results = getName(file, { lower: false });
      res[basename(file)] = results;
    }
    expect(res).toMatchSnapshot();
  });
  it('any w english', async () => {
    const res = {};
    const use = allNonEnglish;
    for (let i = 0; i < use.length; i++) {
      const file = use[i];
      try {
        getName(file);
      } catch (e) {
        expect(e.message).toMatchSnapshot();
      }
    }
  });
  it('any', async () => {
    const res = {};
    const use = allNonEnglish;
    for (let i = 0; i < use.length; i++) {
      const file = use[i];
      const results = await getName(file, { english: false });
      res[basename(file)] = results;
    }
    expect(res).toMatchSnapshot();
  });
});
