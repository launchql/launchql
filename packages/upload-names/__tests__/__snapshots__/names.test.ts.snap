// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uploads any 1`] = `
{
  "–î–º–∏—Ç—Ä–∏–π": "–¥–º–∏—Ç—Ä–∏–π",
  "–õ–µ–≤.—Ä": "–ª–µ–≤.—Ä",
  "„Å©„ÅÜ„ÇÇ„ÅÇ„Çä„Åå„Å®„ÅÜ„Éü„Çπ„Çø„Éº„É≠„Éú„ÉÉ„Éà.txt": "„Å©„ÅÜ„ÇÇ„ÅÇ„Çä„Åå„Å®„ÅÜ„Éü„Çπ„Çø„Éº„É≠„Éú„ÉÉ„Éà.txt",
  "‰Ω†Â•ΩÔºå‰∏ñÁïå": "‰Ω†Â•ΩÔºå‰∏ñÁïå",
  "‰Ω†Â•ΩÔºå‰∏ñÁïå.‰∏ñÁïå": "‰Ω†Â•ΩÔºå‰∏ñÁïå.‰∏ñÁïå",
  "ÂÖ®ÂõΩÊ∏©Ê≥â„Ç¨„Ç§„Éâ.jp": "ÂÖ®ÂõΩÊ∏©Ê≥â„Ç¨„Ç§„Éâ.jp",
}
`;

exports[`uploads double dots are collapsed 1`] = `
{
  "..config.env": ".config.env",
  "archive..tar..gz": "archive.tar.gz",
  "data..v1..report.csv": "data.v1.report.csv",
  "some..file..name.txt": "some.file.name.txt",
}
`;

exports[`uploads english 1`] = `
{
  ".gitignore": ".gitignore",
  ".travis.yml": ".travis.yml",
  "LICENSE": "license",
  "P√ãRSH- - - ENDETJE  BOT√ã": "persh-endetje-bote",
  "P√ãRSHENDETJE BOT√ã": "pershendetje-bote",
  "P√´rshendetje Bot√´": "pershendetje-bote",
  "Selam D√ºnya.jpg": "selam-dunya.jpg",
  "ai.ai": "ai.ai",
  "apng.png": "apng.png",
  "be√ßa": "beca",
  "bmp.bmp": "bmp.bmp",
  "build.js": "build.js",
  "build.js.map": "build.js.map",
  "css.css": "css.css",
  "csv.csv": "csv.csv",
  "docs.js": "docs.js",
  "docs.jsx": "docs.jsx",
  "docx.docx": "docx.docx",
  "dwg.dwg": "dwg.dwg",
  "dxf.dxf": "dxf.dxf",
  "emf.emf": "emf.emf",
  "epub.epub": "epub.epub",
  "font.otf": "font.otf",
  "font.ttf": "font.ttf",
  "font.woff": "font.woff",
  "font.woff2": "font.woff2",
  "gif-w-alpha.gif": "gif-w-alpha.gif",
  "gif.gif": "gif.gif",
  "jpg.jpg": "jpg.jpg",
  "json.json": "json.json",
  "less.less": "less.less",
  "lock.lock": "lock.lock",
  "md.md": "md.md",
  "mp4.mp4": "mp4.mp4",
  "ni√±o.yo": "nino.yo",
  "ods.ods": "ods.ods",
  "odt.odt": "odt.odt",
  "pct.pct": "pct.pct",
  "pdf.pdf": "pdf.pdf",
  "png-w-alpha.png": "png-w-alpha.png",
  "psd.psd": "psd.psd",
  "rtf.rtf": "rtf.rtf",
  "scss.scss": "scss.scss",
  "shellscript": "shellscript",
  "sql.sql": "sql.sql",
  "svg-with-alpha-and-text.svg": "svg-with-alpha-and-text.svg",
  "svg.svg": "svg.svg",
  "svgo.yaml": "svgo.yaml",
  "swf.swf": "swf.swf",
  "tga.tga": "tga.tga",
  "tif.tif": "tif.tif",
  "todo-0.1.0.tgz": "todo-0.1.0.tgz",
  "tsv.tsv": "tsv.tsv",
  "txt.txt": "txt.txt",
  "typescript.ts": "typescript.ts",
  "typescript.tsx": "typescript.tsx",
  "wmf.wmf": "wmf.wmf",
  "xlsx.xlsx": "xlsx.xlsx",
  "zip.zip": "zip.zip",
  "–°–∞–ª–æ–º “∂–∞“≥–æ–Ω.gif": "—Å–∞–ª–æ–º-“∑–∞“≥–æ–Ω.gif",
}
`;

exports[`uploads keeps non-English names when english=false 1`] = `
{
  "„Åì„Çì„Å´„Å°„ÅØ": "„Åì„Çì„Å´„Å°„ÅØ",
  "‰Ω†Â•Ω.txt": "‰Ω†Â•Ω.txt",
  "ÂÖ®ÂõΩÊ∏©Ê≥â„Ç¨„Ç§„Éâ.jp": "ÂÖ®ÂõΩÊ∏©Ê≥â„Ç¨„Ç§„Éâ.jp",
  "üòäüòäüòä.txt": "üòäüòäüòä.txt",
}
`;

exports[`uploads non-ASCII fallback when slug would be empty 1`] = `{}`;

exports[`uploads preserves meaningful dots in name 1`] = `
{
  ".env": ".env",
  "lev.—Ä": "lev.—Ä",
  "my.file.config.json": "my.file.config.json",
  "node.module.js": "node.module.js",
}
`;

exports[`uploads upper 1`] = `
{
  ".gitignore": ".gitignore",
  ".travis.yml": ".travis.yml",
  "LICENSE": "LICENSE",
  "P√ãRSH- - - ENDETJE  BOT√ã": "PERSH-ENDETJE-BOTE",
  "P√ãRSHENDETJE BOT√ã": "PERSHENDETJE-BOTE",
  "P√´rshendetje Bot√´": "Pershendetje-Bote",
  "Selam D√ºnya.jpg": "Selam-Dunya.jpg",
  "ai.ai": "ai.ai",
  "apng.png": "apng.png",
  "be√ßa": "beca",
  "bmp.bmp": "bmp.bmp",
  "build.js": "build.js",
  "build.js.map": "build.js.map",
  "css.css": "css.css",
  "csv.csv": "csv.csv",
  "docs.js": "docs.js",
  "docs.jsx": "docs.jsx",
  "docx.docx": "docx.docx",
  "dwg.dwg": "dwg.dwg",
  "dxf.dxf": "dxf.dxf",
  "emf.emf": "emf.emf",
  "epub.epub": "epub.epub",
  "font.otf": "font.otf",
  "font.ttf": "font.ttf",
  "font.woff": "font.woff",
  "font.woff2": "font.woff2",
  "gif-w-alpha.gif": "gif-w-alpha.gif",
  "gif.gif": "gif.gif",
  "jpg.jpg": "jpg.jpg",
  "json.json": "json.json",
  "less.less": "less.less",
  "lock.lock": "lock.lock",
  "md.md": "md.md",
  "mp4.mp4": "mp4.mp4",
  "ni√±o.yo": "nino.yo",
  "ods.ods": "ods.ods",
  "odt.odt": "odt.odt",
  "pct.pct": "pct.pct",
  "pdf.pdf": "pdf.pdf",
  "png-w-alpha.png": "png-w-alpha.png",
  "psd.psd": "psd.psd",
  "rtf.rtf": "rtf.rtf",
  "scss.scss": "scss.scss",
  "shellscript": "shellscript",
  "sql.sql": "sql.sql",
  "svg-with-alpha-and-text.svg": "svg-with-alpha-and-text.svg",
  "svg.svg": "svg.svg",
  "svgo.yaml": "svgo.yaml",
  "swf.swf": "swf.swf",
  "tga.tga": "tga.tga",
  "tif.tif": "tif.tif",
  "todo-0.1.0.tgz": "todo-0.1.0.tgz",
  "tsv.tsv": "tsv.tsv",
  "txt.txt": "txt.txt",
  "typescript.ts": "typescript.ts",
  "typescript.tsx": "typescript.tsx",
  "wmf.wmf": "wmf.wmf",
  "xlsx.xlsx": "xlsx.xlsx",
  "zip.zip": "zip.zip",
  "–°–∞–ª–æ–º “∂–∞“≥–æ–Ω.gif": "–°–∞–ª–æ–º-“∂–∞“≥–æ–Ω.gif",
}
`;
